packages/core/core-flows/src/definition/cart/steps/validate-variant-prices.ts

```diff
@@ -1,11 +1,13 @@
 import { BigNumberInput } from "@medusajs/types"
-import { MedusaError, isDefined } from "@medusajs/utils"
+import { MedusaError, isPresent } from "@medusajs/utils"
 import { createStep } from "@medusajs/workflows-sdk"
 
 interface StepInput {
   variants: {
     id: string
-    calculated_price?: BigNumberInput
+    calculated_price?: {
+      calculated_amount?: BigNumberInput | null
+    }
```

Comment:
suggestion: I ve create a WithCalculatedPrice in the types package, also I ve used number as once received from the module I believe we have numbers after serialization